framework/db/BaseActiveRecord.php

```diff
@@ -600,10 +600,11 @@ public function markAttributeDirty($name)
      * @param string $name the name of the attribute.
      * @param bool $identical whether the comparison of new and old value is made for
      * identical values using `===`, defaults to `true`. Otherwise `==` is used for comparison.
-     * This parameter is available since version 2.0.4.
      * @return bool whether the attribute has been changed
+     * This parameter was available since version 2.0.4 under the name isAttributeChanged().
+     * @since 3.0.0
      */
-    public function isAttributeChanged($name, $identical = true)
+    public function isAttributeDirty($name, $identical = true)
```

Comment:
Why do you rename this method? I think you should add the only new method `isDirty()`

Reply:
`isAttributeChanged` is better

Reply:
What would be a good reason to have is named isAttributeChanged()?
It is not a matter of what sounds better or what one would prefer or feel more comfortable with.
Here the idea is more to be consistent with the other methods of the object that deal with the same subject like markAttributeDirty(), getDirtyAttributes() and now isDirty().

Consider this example:
```php
//Load an order
$order = Order::findone(1);

//Now, don't change the value but mark it as 'dirty'
$order->markAttributeDirty('quantity');

//Check if the attribute is dirty
if ($order->isAttributeDirty('quantity')) {
    //...
}
```
Doesn't it look pretty fluent?

Reply:
The first reason is BC.

Reply:
You are right, but this is proposed in 3.0.0 where BC exist.
And it could be added and marked as deprecated in 2.0.16.
BC should not be a reason to keep burdens.